src/core/frame/names.cc

```diff
@@ -351,6 +351,24 @@ void Frame::_init_names(XTypeMaker& xt) {
 // Options
 //------------------------------------------------------------------------------
 
+static const char * doc_options_frame_names_auto_index =
+R"(
+When Frame needs to auto-name columns, they will be assigned
+names `C0`, `C1`, `C2`, etc. by default. This option allows you to
+control the starting index in this sequence. For example, setting
+`dt.options.frame.names_auto_index=1` will cause the columns to be
+named `C1`, `C2`, `C3`, etc.
```

Comment:
I would add a see-also link to `manual/name_mangling`: 
```
:ref:`name-mangling`
```

Reply:
Added this section for both frame options.